### ISSUE TITLE
 Investigate the possibility to change the code of API responses with WazuhClusterError to 500

### ISSUE THREAD

**Issue by mcarmona99** [wazuh/wazuh#10364]
Right now, we are showing the exceptions `WazuhClusterError` with code **400** in the API responses (Bad Request).

We should investigate the possibility to change these codes from **400** to **500** as the `WazuhClusterError` may correspond to a server-side error instead of a bad request. In this issue, we should also see the possible implications and study whether every WazuhClusterError should appear with code 500 or not.

For instance, we can see this `WazuhClusterError` that should be, in truth, a **500** code error. When it is at the API level, it is created as a "problem" with code **400**.

https://github.com/wazuh/wazuh/blob/18b698f7f7600568e48e5bcc1ff9738d2392c63a/framework/wazuh/core/cluster/control.py#L55-L58

A possible solution to show `WazuhClusterError` with code **500** could be adding the specific exception to the `if/else` blocks in the `_create_problem` API function. Right now the `WazuhClusterError` is raised with code **400** because it is a child of the `WazuhError` class.

https://github.com/wazuh/wazuh/blob/18b698f7f7600568e48e5bcc1ff9738d2392c63a/api/api/util.py#L240-L278

### wazuh/wazuh
- Unit tests without failures. Updated and/or expanded if there are new functions/methods/outputs:
  - [x] Cluster (`framework/wazuh/core/cluster/tests/` & `framework/wazuh/core/cluster/dapi/tests/`)
  - [x] Core (`framework/wazuh/core/tests/`)
  - [x] SDK (`framework/wazuh/tests/`)
  - [x] RBAC (`framework/wazuh/rbac/tests/`)
  - [x] API (`api/api/tests/`)
- API tavern integration tests without failures. Updated and/or expanded if needed (`api/test/integration/`):
  - [x] Affected tests 
  - [x] Affected RBAC (black and white) tests
- [x] Review integration test mapping using the script (`api/test/integration/mapping/integration_test_api_endpoints.json`)
- [x] Review of spec.yaml examples and schemas (`api/api/spec/spec.yaml`)
- [x] Review exceptions remediation when any endpoint path changes or is removed (`framework/wazuh/core/exception.py`)
- [x] Changelog (`CHANGELOG.md`)

### wazuh/wazuh-documentation
- [x] Migration from 3.X for changed endpoints (`source/user-manual/api/equivalence.rst`)
- [x] Update RBAC reference with new/modified actions/resources/relationships (`source/user-manual/api/rbac/reference.rst`)

**Comment by mcarmona99** [comment]
## Issue update


The HTTP responses are grouped in five classes, depending on the status code:

1. Informational responses (100–199)
2. Successful responses (200–299)
3. Redirection messages (300–399)
4. Client error responses (400–499)
5. Server error responses (500–599)

Ref: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status

Right now, the `WazuhClusterError` exceptions create API problems with status code 400 (this is done in the `api/util.py` file). As it was mentioned in the issue's body, the status code should not be 400 as this error is not a client error.

According to the [Mozilla documentation about the HTTP protocol](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400), the `400 Bad Request` response status code indicates that the server cannot or will not process the request due to something that is perceived to be a client error (for example, malformed request syntax, invalid request message framing, or deceptive request routing).

According to the [Mozilla documentation about the HTTP protocol](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/500), the `500 Internal Server Error` server error response code indicates that the server encountered an unexpected condition that prevented it from fulfilling the request.

#### When do we raise a `WazuhClusterError`?

- `get_nodes`, `get_node`, `get_health` and `get_agents` functions of `framework/wazuh/core/cluster/control.py`: The `WazuhClusterError` is raised in these functions when, after communicating with the cluster, the response cannot be loaded properly (`JSONDecodeError`). For these cases, the error is raised internally due to cluster communications, and therefore, the error is a server-side one (500).

- `process_request` and `send_file_request` methods of `LocalServerHandlerMaster` class (`framework/wazuh/core/cluster/local_server.py`): When the request/file is tried to be forwarded to the cluster and the node ID is unknown. Both cases should be treated as server errors too.

- `process_request`, `send_request_to_master`, and `send_file_request` methods of LocalServerHandlerWorker class (`framework/wazuh/core/cluster/local_server.py`): when the worker node is not connected to the master node. This error does not depend on the user as it is related to the dapi communications. Again server error. This case is similar to the last one: the last case was master -> worker, and this one is worker -> master.

- `process_request` method of `WorkerHandler` class (`framework/wazuh/core/cluster/worker.py`): when a `MemoryError` or a generic `Exception` is raised when trying to send a request to the cluster (from worker). This error is also a server-side one.

- `execute` and `process_dapi_res` methods of `MasterHandler` class (`framework/wazuh/core/cluster/master.py`): error raised due to cluster communications: unknown node ID, timeouts when waiting for responses, etc. Server-side error.


- `sync_worker_files` method of `MasterHandler` class (`framework/wazuh/core/cluster/master.py`): error while processing extra-valid files. Server-side error.

- `get_messages` and `data_received` methods of `Handler` class (`framework/wazuh/core/cluster/common.py`): error when trying to decrypt a cluster message. Server-side error.

- `get_chunks_in_task_id` method of `Handler` class (`framework/wazuh/core/cluster/common.py`): `ValueError` or `KeyError` when loading received messages. Server side error.

- `update_chunks_wdb` method of `Handler` class (`framework/wazuh/core/cluster/common.py`): error while processing agent-info chunks. Server-side error.

- `end_receiving_file` method of `Handler` class (`framework/wazuh/core/cluster/common.py`): error when trying to receive the task: unknown task name. Server-side error.

- `sync` method of `SyncFiles` class, and `sync` method of `SyncWazuhdb` class (`framework/wazuh/core/cluster/common.py`): error response from cluster. Server-side error.


All of the above are `WazuhClusterError` occurrences that may correspond to a response with status code **500**.



- `hello` method of `AbstractServerHandler` class (`framework/wazuh/core/cluster/server.py`): when the worker node ID already exists or the name is the same as the master's. In this case, we could consider the API response as 400 status code's. This error depends on the user as changing the conflicting node name solves the issue.

- `hello` method of `MasterHandler` class (`framework/wazuh/core/cluster/master.py`): when the cluster name is not the excepted one or the worker and master versions are different. Similar to the `hello` method commented above, this error could be a client-side error as a proper configuration solves the issue.

For these 2 occurrences, they could be interpreted as client-side errors, as I said, but the truth is that responses with code 400 indicate a bad request, as was commented above. A bad request indicates that the server cannot or will not process the request due to something that is perceived to be a client error. For these 2 cases, it is true that the existence of the errors can depend on the user, but the server could process the request as there were no errors from the client-side. The errors were generated when processing the request on the server-side.

To sum up, after investigating all the possible `WazuhClusterError` exceptions, all of them should correspond to an API response with code **500**, as they are server-side errors.
